### PR TITLE
dependency: update to angular 11

### PR DESCRIPTION
**Pull request by jma**
Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

## Why are you opening this PR?

- Angular 8 is end of life.

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: projects/rero/ng-core/src/lib/menu/menu-item.ts [hunk 1]
```diff
@@ -138,7 +138,7 @@ export class MenuItem implements ItemInterface {
    * @return boolean
    */
   hasRouterLink(): boolean {
-    return isArray(this._routerLink) && this._routerLink.length > 0
+    return Array.isArray(this._routerLink) && this._routerLink.length > 0
     ? true
     : false;
```
Comment (zannkukai) [inline projects/rero/ng-core/src/lib/menu/menu-item.ts:143]: Note : We could simplify with only `return (Array.isArray(this._routerLink) && this._routerLink.length > 0);`

File: projects/rero/ng-core/src/lib/record/editor/widgets/load-template-form/load-template-form.component.ts [hunk 2]
```diff
@@ -52,21 +55,18 @@ export class LoadTemplateFormComponent implements OnInit {
...
-      private _toastrService: ToastrService,
-      private _route: ActivatedRoute
+      private _toastrService: ToastrService
   ) { }
 
   /**
    * hook OnInit
    */
   ngOnInit() {
-    const initialState: any = this._modalService.config.initialState;
```
Comment (zannkukai) [inline projects/rero/ng-core/src/lib/record/editor/widgets/load-template-form/load-template-form.component.ts:69]: I learn something 👍

File: projects/rero/ng-core/src/lib/translate/date-translate-pipe.ts [hunk 3]
```diff
@@ -34,16 +34,14 @@ export class DateTranslatePipe extends DatePipe implements PipeTransform {
...
+  transform(value: Date | string | number, format?: string, timezone?: string, locale?: string): string | null;
+  transform(value: null | undefined, format?: string, timezone?: string, locale?: string): null;
+  transform(value: Date | string | number | null | undefined, format?: string, timezone?: string, locale?: string): string | null
+  {
     if (!locale) {
       locale = this._translateService.currentLanguage;
     }
 
-    return super.transform(value, format, timezone, locale);
+    return super.transform(value);
```
Comment (zannkukai) [inline projects/rero/ng-core/src/lib/translate/date-translate-pipe.ts:45]: Why removing `format, timezone, locale` from parameters ? There seem still valid in Angular11

File: projects/ng-core-tester/src/app/service/app-menu.service.spec.ts [hunk 4]
```diff
@@ -17,7 +17,7 @@
 
 /* tslint:disable:no-unused-variable */
 
-import { TestBed, async, inject } from '@angular/core/testing';
+import { TestBed, inject, waitForAsync } from '@angular/core/testing';
```
Comment (sebastiendeleze) [inline projects/ng-core-tester/src/app/service/app-menu.service.spec.ts:20]: new import is not used

File: projects/rero/ng-core/src/lib/record/editor/editor.component.ts [hunk 5]
```diff
@@ -40,6 +40,9 @@ import { isEmpty, orderedJsonSchema, removeEmptyValues } from './utils';
 import { LoadTemplateFormComponent } from './widgets/load-template-form/load-template-form.component';
 import { SaveTemplateFormComponent } from './widgets/save-template-form/save-template-form.component';
 
+export interface JSONSchema7 extends JSONSchema7Base {
+  form: any;
+}
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/editor/editor.component.ts:45]: maybe a space after the interface, as this is not linked to component. Or better in a specific file.

File: projects/rero/ng-core/src/lib/record/editor/widgets/load-template-form/load-template-form.component.ts [hunk 6]
```diff
@@ -41,6 +41,9 @@ export class LoadTemplateFormComponent implements OnInit {
   model: FormModel;
 
 
+  templateResourceType: string;
+  resourceType: string;
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/editor/widgets/load-template-form/load-template-form.component.ts:45]: Add a comment for each properties

File: projects/rero/ng-core/src/lib/translate/date-translate-pipe.ts (hunk 3 above)
Comment (sebastiendeleze) [reply projects/rero/ng-core/src/lib/translate/date-translate-pipe.ts:45]: +1

File: projects/rero/ng-core/src/lib/translate/date-translate-pipe.ts [hunk 7]
```diff
@@ -34,12 +34,10 @@ export class DateTranslatePipe extends DatePipe implements PipeTransform {
...
 
-  transform(
-    value: any,
-    format = 'mediumDate',
-    timezone?: string,
-    locale?: string
-  ): string {
+  transform(value: Date | string | number, format?: string, timezone?: string, locale?: string): string | null;
+  transform(value: null | undefined, format?: string, timezone?: string, locale?: string): null;
+  transform(value: Date | string | number | null | undefined, format?: string, timezone?: string, locale?: string): string | null
```
Comment (Garfield-fr) [inline projects/rero/ng-core/src/lib/translate/date-translate-pipe.ts:39]: Why 3 times the same function ? Only one needed

**Review by Garfield-fr — CHANGES_REQUESTED** [review]

**Review by sebastiendeleze — APPROVED** [review]

File: projects/rero/ng-core/src/lib/translate/date-translate-pipe.ts (hunk 7 above)
Comment (jma) [reply projects/rero/ng-core/src/lib/translate/date-translate-pipe.ts:39]: This is required as this function is inherited from the parent pipe.

Comment (Garfield-fr) [reply projects/rero/ng-core/src/lib/translate/date-translate-pipe.ts:39]: Thanks for the precision 👍🏻

**Review by Garfield-fr — APPROVED** [review]

**Pull request merged**